proto/qdrant/cloud/event/v1/events.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+
+package qdrant.cloud.event.v1;
+
+import "buf/validate/validate.proto";
+import "google/protobuf/descriptor.proto";
+import "google/protobuf/timestamp.proto";
+import "qdrant/cloud/common/v1/common.proto";
+
+// The method options for configuring the event options.
+// If this option is not found for an RPC, no event will be generated.
+//
+// Example for a CreateBackup RPC:
+//  option (qdrant.cloud.event.v1.event_options) = {
+//    event_type: qdrant.cloud.event.v1.EVENT_TYPE_CREATED,
+//    resource_type: "backup",
+//    resource_id_field: "resp.backup.id",
+//    resource_url_template: "/Accounts/{req.backup.account_id}/Clusters/{req.backup.cluster_id}/Backups/{resp.backup.id}",
+//    additional_context_fields: {
+//      key: "cluster_id"
+//      value: "req.cluster_id"
+//    }
+//  };
+//
+extend google.protobuf.MethodOptions {
+  // The actual event options.
+  EventOptions event_options = 51001;
+}
+
+// EventOptions is a custom method option to indicate that an event should be
+// generated when this RPC is successfully called.
+// The fields should start with 'req.' or 'resp.' to indicate whether the info needs to be
+// extracted from the request or the response.
+message EventOptions {
+  // The type of the event that should be generated.
+  EventType event_type = 1 [(buf.validate.field).enum = {
+    defined_only: true
+    not_in: [0]
+  }];
+  // The type of the resource that this event is about (e.g., "cluster", "backup").
+  string resource_type = 2 [(buf.validate.field).string.min_len = 1];
+  // Whether or not this event should be marked as a status only event.
+  // TODO: Alternative we can make EVENT_TYPE_UPDATED_SPEC and EVENT_TYPE_UPDATED_STATUS, in my opinion this bool makes more sense.
```

Comment:
At a first glance I like `EVENT_TYPE_UPDATED_SPEC` more than the boolean. Thinking also about subscribing to those events, feels more natural to pass a list of event_types you want to filter out.